src/tactic/lint/misc.lean

```diff
@@ -448,3 +448,67 @@ meta def linter.unused_haves_suffices : linter :=
 "`proof_of_goal`, in addition to being ineffectual, they may make unnecessary assumptions in " ++
 "proofs appear as if they are used. ",
   is_fast := ff }
+
+/-!
+## Linter for equalities and iff's
+-/
+
+/--
+Recursively consumes a Pi expression while accumulating names and de-Bruijn indexes of explicit
+variables, ultimately obtaining the remaining non-Pi expression as well.
+-/
+meta def unravel_explicits_of_pi :
+  expr → ℕ → list name → list ℕ → (list name) × (list ℕ) × expr
+| (pi n bi _ e) i ln li :=
+  match bi with
+  | binder_info.default := unravel_explicits_of_pi e (i + 1) (ln.concat n) (li.concat i)
```

Comment:
```suggestion
| (pi n binder_info.default _ e) i ln li :=
  unravel_explicits_of_pi e (i + 1) (ln.concat n) (li.concat i)
| (pi n _ _ _ e) i ln li :=
  unravel_explicits_of_pi e (i + 1) ln li
| e             _ ln li := (ln, li, e)
```
something like this, nesting the matches inline without explicitly doing a nested match should work